DEVELOPING_OPAL/tools/src/main/scala/org/opalj/support/info/StringAnalysisReflectiveCalls.scala

```diff
@@ -0,0 +1,311 @@
+ * string values are / could be passed to these calls.
+ * <p>
+ * Currently, this runner supports / handles the following reflective calls:
+ * <ul>
+ * <li>`Class.forName(string)`</li>
+ * <li>`Class.forName(string, boolean, classLoader)`</li>
+ * <li>`Class.getField(string)`</li>
+ * <li>`Class.getDeclaredField(string)`</li>
+ * <li>`Class.getMethod(String, Class[])`</li>
+ * <li>`Class.getDeclaredMethod(String, Class[])`</li>
+ * </ul>
+ *
+ * @author Patrick Mell
+ */
+object StringAnalysisReflectiveCalls extends ProjectAnalysisApplication {
+
+    private type ResultMapType = mutable.Map[String, ListBuffer[StringConstancyInformation]]
+
+    /**
+     * Stores a list of pairs where the first element corresponds to the entities passed to the
+     * analysis and the second element corresponds to the method name in which the entity occurred,
+     * i.e., a value in [[relevantMethodNames]].
+     */
+    private val entityContext = ListBuffer[(P, String)]()
+
+    /**
+     * Stores all relevant method names of the Java Reflection API, i.e., those methods from the
+     * Reflection API that have at least one string argument and shall be considered by this
+     * analysis. The string are supposed to have the format as produced by [[buildFQMethodName]].
+     */
+    private val relevantMethodNames = List(
+        // The following is for the javax.crypto API
+        //"javax.crypto.Cipher#getInstance", "javax.crypto.Cipher#getMaxAllowedKeyLength",
+        //"javax.crypto.Cipher#getMaxAllowedParameterSpec", "javax.crypto.Cipher#unwrap",
+        //"javax.crypto.CipherSpi#engineSetMode", "javax.crypto.CipherSpi#engineSetPadding",
+        //"javax.crypto.CipherSpi#engineUnwrap", "javax.crypto.EncryptedPrivateKeyInfo#getKeySpec",
+        //"javax.crypto.ExemptionMechanism#getInstance", "javax.crypto.KeyAgreement#getInstance",
+        //"javax.crypto.KeyGenerator#getInstance", "javax.crypto.Mac#getInstance",
+        //"javax.crypto.SealedObject#getObject", "javax.crypto.SecretKeyFactory#getInstance"
+        // The following is for the Java Reflection API
+        "java.lang.Class#forName", "java.lang.ClassLoader#loadClass",
+        "java.lang.Class#getField", "java.lang.Class#getDeclaredField",
+        "java.lang.Class#getMethod", "java.lang.Class#getDeclaredMethod"
+    )
+
+    /**
+     * A list of fully-qualified method names that are to be skipped, e.g., because they make an
+     * analysis crash (e.g., com/sun/jmx/mbeanserver/MBeanInstantiator#deserialize)
+     */
+    private val ignoreMethods = List()
+
+    // executeFrom specifies the index / counter when to start feeding entities to the property
+    // store. executeTo specifies the index / counter when to stop feeding entities to the property
+    // store. These values are basically to help debugging. executionCounter is a helper variable
+    // for that purpose
+    private val executeFrom = 0
```

Comment:
From the description, I don't get the purpose of these vals. What is counted? Please adjust the documentation, or remove them, if they are only used for debug purposes.

Reply:
Removed in https://github.com/opalj/opal/pull/1/commits/38bfe45ea1421e9f452297641b015a75737880fa